generics/helpers/common_handler_v2.js

```diff
@@ -0,0 +1,848 @@
+							// Update the leastStartDate with the earliest task start date
+							leastStartDate = eachTask.startDate
+						}
+						labels.push(eachTask.title)
+						data.push({
+							// Add task details to data array
+							task: eachTask.title,
+							startDate: eachTask.startDate,
+							endDate: eachTask.endDate,
+						})
+					})
+				)
+				// Determine the actual least start date among tasks
+				if (data.length > 0) {
+					data.forEach((v) => {
+						leastStartDate = new Date(v.startDate) < new Date(leastStartDate) ? v.startDate : leastStartDate
+					})
+				}
+
+				// Define chart options for the project
+				let chartOptions = {
+					order: 1,
+					options: {
+						type: 'horizontalBar',
+						data: {
+							labels: labels,
+							datasets: [
+								{
+									// Dataset for start dates relative to leastStartDate
+									data: data.map((t) => {
+										if (leastStartDate && t.startDate) {
+											return dateDiffInDays(new Date(leastStartDate), new Date(t.startDate))
+										}
+										return null
+									}),
+									datalabels: {
+										color: '#025ced',
+										//   formatter: function (value, context) {
+										//     return '';
+										//   },
+									},
+									backgroundColor: 'rgba(63,103,126,0)',
+									hoverBackgroundColor: 'rgba(50,90,100,0)',
+								},
+								{
+									// Dataset for task durations (end dates - start dates)
+									data: data.map((t) => {
+										if (t.startDate && t.endDate) {
+											return dateDiffInDays(new Date(t.startDate), new Date(t.endDate))
+										}
+									}),
+									datalabels: {
+										color: '#025ced',
+										//   formatter: function (value, context) {
+										//     return '';
+										//   },
+									},
+								},
+							],
+						},
+						options: {
+							maintainAspectRatio: false,
+							title: {
+								display: true,
+								text: eachProject.projectDetails[0].title,
+							},
+							legend: { display: false },
+							scales: {
+								x: {
+									stacked: true,
+									ticks: {
+										// Custom tick formatter to display date labels
+										callback: function (value, index, values) {
+											if (leastStartDate) {
+												const date = new Date(leastStartDate)
+												date.setDate(value)
+												return getDate(date)
+											}
+										},
+									},
+								},
+
+								y: {
+									stacked: true,
+								},
+							},
+						},
+					},
+				}
+				// Store chart options in the array
+				arrayOfChartData.push(chartOptions)
+				eachProject.order = i
+				projectData.push(eachProject)
+				i++
+			})
+		)
+
+		resolve([arrayOfChartData, projectData])
+	})
+}
+
+/**
+ * Calculate the difference in days between two dates.
+ * @param {Date} a - The first date.
+ * @param {Date} b - The second date.
+ * @returns {number} The difference in days between the two dates (b - a).
+ */
+function dateDiffInDays(startDate, endDate) {
```

Comment:
@MallanagoudaB Why duplicate this function? once you add the function to utils you can call it directly where ever it is used.